### PR TITLE
Feature/enable reductions stack

### PR DESCRIPTION
**Pull request by ivan-pavlov**
This PR adds new `add_reduction` function.
I looked closer into how VW [stacks reductions](https://github.com/JohnLangford/vowpal_wabbit/blob/a9bba04b13bb87878c6d24ef212c220a7cb4822a/vowpalwabbit/parse_args.cc#L1089-L1150) and decided to add this functionality as it sounds like a natural way for VW. 

You can work with diffrent reductions like that:
```
test_vwmodel <-  vwsetup(dir = "./", model = "mdl.vw",
                         reduction = "ect", num_classes=3) %>%
    add_reduction(reduction = "boosting", num_learners=10)
```

In this way, you can add multiple reductions to  VW model.

### CODE REVIEW

**Comment by eddelbuettel** [comment]
What is a 'reduction' ? Is there VW documentation for us to read up on?

**Comment by ivan-pavlov** [comment]
This is how VW calls different learning modes:
https://github.com/JohnLangford/vowpal_wabbit/wiki/Command-line-arguments#learning-algorithm--reduction-options

But I think now, maybe word "reductions" is not general enough.

**Comment by eddelbuettel** [comment]
Nice.  A demo file might illustrate this well.

**Comment by ivan-pavlov** [comment]
I have an example on how to use this in `demo/example_1.R` file, but I am also planning to add more examples with different learning modes and reductions.

File: tests/testthat/test-vwsetup.R [hunk 1]
```diff
@@ -36,18 +34,18 @@ test_model <- list(params = list(learning_mode = "binary",
...
-                                   quantile_tau=0.5)
+                                   quantile_tau=0.5),
+                                 reductions = list()
 
 ),
 dir = "../my_tmp/",
 model = "mdl.vw",
-update_model = TRUE,
-params_str = paste("  --passes 1 --bit_precision 18 --holdout_period 10",
+params_str = paste(" --passes 1 --bit_precision 18 --holdout_period 10",
```
Comment (coatless) [inline tests/testthat/test-vwsetup.R]: Should be `paste0(...)` instead of `paste(... ,sep = "")`

File: tests/testthat/test-vwsetup.R [hunk 2]
```diff
@@ -61,45 +59,43 @@ test_that("vwsetup correctly setup model with different learning modes", {
...
+  lda_test_model$params$reductions = list(lda = list(num_topics=0,
                                            lda_alpha=0.100000001,
                                            lda_rho=0.100000001,
                                            lda_D=10000,
                                            lda_epsilon=0.00100000005,
                                            math_mode=0,
-                                           minibatch=1)
-  lda_test_model$params_str = paste("--lda 0  --passes 1 --bit_precision 18",
+                                           minibatch=1))
+  lda_test_model$params_str = paste(" --passes 1 --bit_precision 18",
```
Comment (coatless) [inline tests/testthat/test-vwsetup.R]: Ditto on `paste0(...)`

File: tests/testthat/test-vwsetup.R [hunk 3]
```diff
@@ -108,19 +104,19 @@ test_that("vwsetup correctly setup model with different learning modes", {
   custom_test_model$params$general_params$cache = TRUE
   custom_test_model$params$general_params$passes = 10
   custom_test_model$params$optimization_params$adaptive = FALSE
-  custom_test_model$params$learning_params$binary = TRUE
-  custom_test_model$params_str <- paste("  --passes 10 --bit_precision 18",
+  custom_test_model$params$reductions = list(binary = list())
+  custom_test_model$params_str <- paste(" --passes 10 --bit_precision 18",
```
Comment (coatless) [inline tests/testthat/test-vwsetup.R]: Ditto on `paste0(...)`

**Review by coatless — CHANGES_REQUESTED** [review]

File: DESCRIPTION [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Package: rvwgsoc
 Type: Package
 Title: R interface for Vowpal Wabbit
-Version: 0.1
+Version: 0.2
 Date: 2018-05-22
```
Comment (eddelbuettel) [inline DESCRIPTION:5]: Try to remember to update the date with the version.  (Ie next time you do.)

**Review by eddelbuettel — APPROVED** [review]
Looks good to me (have not run it yet though).

**Review by coatless — APPROVED** [review]

**Pull request merged**